### DISCUSSION TITLE
"Blocked aria-hidden on a <body> element" error when using `Dialog`

### DISCUSSION THREAD

**Discussion by ivancuric** [chakra-ui/ark#2717] [General]
### Description

When using the `defaultOpen` prop on the `Dialog` element, Chrome reports the following:

> Blocked aria-hidden on a `<body>` element because it would hide the entire accessibility tree from assistive technology users. For more details, see the aria-hidden section of the WAI-ARIA specification at [https://w3c.github.io/aria/#aria-hidden](https://w3c.github.io/aria/#aria-hidden). 

This does not seem to happen when opening a modal on a user interaction.

Interestingly, testing the demo on https://ark-ui.com/react/docs/components/dialog, I get a different message:
> Blocked aria-hidden on a `<button>` element because the element that just received focus must not be hidden from assistive technology users. Avoid using aria-hidden on a focused element or its ancestor. Consider using the inert attribute instead, which will also prevent focus. For more details, see the aria-hidden section of the WAI-ARIA specification at [https://w3c.github.io/aria/#aria-hidden](https://w3c.github.io/aria/#aria-hidden). 

Same happens when interacting with the framework selection dropdown, but for a `div` element.

### Link to Reproduction (or Detailed Explanation)

https://stackblitz.com/edit/ark-dialog-shadow-kzuhk2?file=src%2FApp.tsx

### Steps to Reproduce

Go to the repro site and observe the console.

### Ark UI Version

3.6.2

### Framework

- [ ] React
- [X] Solid
- [ ] Vue

### Browser

Chrome

**Comment by ivancuric** [comment]
As far as I see, [`@zag-js/aria-hidden`](https://github.com/chakra-ui/zag/tree/15e546edf1e088da597ffc1de8d965147b563cc8/packages/utilities/aria-hidden) could be tweaked to only apply the `inert` attribute instead of `aria-hidden`. Am I correct? 

Also, interestingly, `aria-hidden` isn't applied in this example:
https://stackblitz.com/edit/vitejs-vite-yccjeg?file=src%2FApp.tsx

**Comment by ivancuric** [comment]
Why was this converted to a discussion?

**Comment by segunadebayo** [comment]
Closing — Dialog now handles `aria-hidden` safely. This has been fixed.
